src/opensearch.py

```diff
@@ -67,6 +70,15 @@ def install(self):
             logger.error(f"Failed to install opensearch. \n{e}")
             raise OpenSearchInstallError()
 
+        # In opensearch snap, CA truststore should be placed in another folder than JDK's
+        # default path. That happens because JDK sits in the /snap folder and is read-only.
+        # Run this setup once, in the charm's lifecycle, as the truststore should land in
+        # the SNAP_COMMON folder.
+        ca_truststore = OpenSearchCATruststore(self._charm)
+        self.set_truststore_file()
+        # Trusstore has been created, then we need to add configuration to jvm.options
+        self._charm.opensearch_config.configure_jvm_ca_truststore(ca_truststore.get_jvm_config())
```

Comment:
I'm still not sure why we need the JVM trust store. We can create / use our own.

Reply:
Actually, we are using it by default whenever we spin up the JVM. What I am doing there is making a copying and adding one extra cert if we have a self-signed cert. Then, in that case, I need to tell the JVM to get its default certs from another store